test/unit/product/product-component.js

```diff
@@ -3,715 +3,926 @@ import Checkout from '../../../src/components/checkout';
+              value: true,
+            });
           });
-          product.modal = null;
-          product.props.createModal = createModalStub;
-        });
 
-        it('creates Modal and initializes modal with model', () => {
-          product.openModal();
-          assert.calledWith(createModalStub, sinon.match.object, product.props);
-          assert.calledWith(initSpy, product.model);
-        });
-      });
-    });
+          afterEach(() => {
+            if (variantForOptionsStub.restore) {
+              variantForOptionsStub.restore();
+            }
+          });
 
-    describe('setDefaultVariant()', () => {
-      it('sets selectedVariant\'s id to product.defaultVariantId', () => {
-        product.defaultStorefrontVariantId = 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xMjM0Nw==';
-        product.setDefaultVariant(productFixture);
-        assert.equal(product.selectedOptions.Print, 'shark');
-        assert.equal(product.selectedOptions.Size, 'large');
-      });
+          it('caches image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({image: 'test-image'});
+            product.updateVariant('Size', 'large');
+            assert.equal(product.cachedImage, 'test-image');
+          });
 
-      it('falls back to first variantId if invalid variantId was provided', () => {
-        product.defaultStorefrontVariantId = 'this is an invalid variant id';
-        product.setDefaultVariant(productFixture);
-        assert.equal(product.selectedOptions.Print, 'sloth');
-        assert.equal(product.selectedOptions.Size, 'small');
-      });
-    });
+          it('sets the selected image to null if selected variant has an image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({image: 'test-image'});
+            product.updateVariant('Size', 'large');
+            assert.isNull(product.selectedImage);
+          });
 
-    describe('getters', () => {
-      describe('shouldUpdateImage', () => {
-        describe('if no cached image', () => {
-          it('returns true', () => {
-            product.cachedImage = null;
-            assert.ok(product.shouldUpdateImage);
+          it('sets the selected image to the first image in model if selected variant does not have an image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({});
+            product.updateVariant('Size', 'large');
+            assert.equal(product.selectedImage, product.model.images[0]);
           });
         });
 
-        describe('if image and cached image are different', () => {
-          beforeEach(async () => {
-            product.config.product.width = '100px';
-            await product.init(testProductCopy);
+        it('sets selected image to the cached image in model if variant does not exist', () => {
```

Comment:
I think this test needs to be outside of the above `if variant exists` `describe` block?

Reply:
the test seems to already exist outside of the 'if variant exists' block